api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -150,6 +154,8 @@ type MachineConfigPool struct {
 	// +optional
 	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Optional configuration enforced on this NodeGroup"
 	Config *NodeGroupConfig `json:"config,omitempty"`
+	// NodeGroupName the name of the node group this MCP belongs to
+	NodeGroupName string `json:"nodeGroupName,omitempty"`
```

Comment:
ok, this is generated by the system so if we get this far it MUST be present. But still we need to mark as optional because it was not present in the earlier v1 API, so a well-behaving client is not required to consume it. Hence must be optional even if in practice it will always be present starting 4.18